### ISSUE TITLE
test TestCtlV3WatchClientTLS

### ISSUE THREAD

**Issue by xiang90** [coreos/etcd#8739]
```
--- FAIL: TestCtlV3WatchClientTLS (15.11s)
	ctl_v3_watch_test.go:75: watchTest #3-0: ctlV3Put error (read /dev/ptmx: input/output error (expected "OK", got ["Error: dial tcp 127.0.0.1:20000: getsockopt: connection refused\r\n"]))
	ctl_v3_watch_test.go:82: watchTest #3: ctlV3Watch error (read /dev/ptmx: input/output error)
	testutil.go:55: goroutine 3502 [running]:
		github.com/coreos/etcd/pkg/testutil.FatalStack(0xc42022a000, 0xc42019a840, 0x18)
			/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/pkg/testutil/testutil.go:54 +0x74
		github.com/coreos/etcd/e2e.testCtl(0xc42022a000, 0xcc10b0, 0xc420038f90, 0x1, 0x1)
			/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/e2e/ctl_v3_test.go:164 +0x483
		github.com/coreos/etcd/e2e.TestCtlV3WatchClientTLS(0xc42022a000)
			/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/e2e/ctl_v3_watch_test.go:24 +0xa9
		testing.tRunner(0xc42022a000, 0xcc0ca8)
			/usr/local/go/src/testing/testing.go:746 +0xd0
		created by testing.(*T).Run
			/usr/local/go/src/testing/testing.go:789 +0x2de
		
		goroutine 1 [chan receive]:
		testing.(*T).Run(0xc42022a0f0, 0xc9d96b, 0x17, 0xcc0ca8, 0x48f501)
			/usr/local/go/src/testing/testing.go:790 +0x2fc
		testing.runTests.func1(0xc42022a0f0)
			/usr/local/go/src/testing/testing.go:1004 +0x64
		testing.tRunner(0xc42022a0f0, 0xc420221d98)
			/usr/local/go/src/testing/testing.go:746 +0xd0
		testing.runTests(0xc4201824e0, 0x10af740, 0x97, 0x97, 0xc4200187f7)
			/usr/local/go/src/testing/testing.go:1002 +0x2d8
		testing.(*M).Run(0xc420221f18, 0xc9d674)
			/usr/local/go/src/testing/testing.go:921 +0x111
		github.com/coreos/etcd/e2e.TestMain(0xc420221f18)
			/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/e2e/main_test.go:52 +0x4d5
		main.main()
			github.com/coreos/etcd/e2e/_test/_testmain.go:342 +0xdb
		
		goroutine 38 [chan receive]:
		github.com/coreos/etcd/pkg/logutil.(*MergeLogger).outputLoop(0xc420188be0)
			/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/pkg/logutil/merge_logger.go:174 +0x47a
		created by github.com/coreos/etcd/pkg/logutil.NewMergeLogger
			/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/pkg/logutil/merge_logger.go:92 +0xb8
		
		goroutine 3485 [chan receive]:
		github.com/coreos/etcd/e2e.watchTest(0xc42022a000, 0xc420016b00, 0xb, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, 0x4e20, ...)
			/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/e2e/ctl_v3_watch_test.go:85 +0x5fe
		github.com/coreos/etcd/e2e.testCtl.func2(0xc42001e360, 0xcc10b0, 0xc42022a1e0)
			/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/e2e/ctl_v3_test.go:155 +0x83
		created by github.com/coreos/etcd/e2e.testCtl
			/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/e2e/ctl_v3_test.go:153 +0x2c5
		
	testutil.go:56: test timed out after 15s
	ctl_v3_test.go:148: error closing etcd processes (exit status 2)
```

**Comment by gyuho** [comment]
@xiang90 Duplicate with https://github.com/coreos/etcd/issues/8627. Which is fixed in grpc-go >v1.7.

I can reproduce with current master branch:

```
=== RUN   TestCtlV3WatchClientTLS
../bin/etcd-23083: 2017-10-25 13:50:14.586668 W | pkg/flags: unrecognized environment variable ETCD_UNSUPPORTED_ARCH=amd64
../bin/etcd-23083: 2017-10-25 13:50:14.586736 I | etcdmain: etcd Version: 3.2.0+git
../bin/etcd-23083: 2017-10-25 13:50:14.586756 I | etcdmain: Git SHA: fff1fb2
../bin/etcd-23083: 2017-10-25 13:50:14.586759 I | etcdmain: Go Version: go1.9.1
../bin/etcd-23083: 2017-10-25 13:50:14.586779 I | etcdmain: Go OS/Arch: linux/amd64
../bin/etcd-23083: 2017-10-25 13:50:14.586786 I | etcdmain: setting maximum number of CPUs to 8, total number of available CPUs is 8
../bin/etcd-23083: 2017-10-25 13:50:14.587009 I | embed: listening for peers on http://localhost:20001
../bin/etcd-23083: 2017-10-25 13:50:14.587060 I | embed: listening for client requests on localhost:20000
../bin/etcd-23083: 2017-10-25 13:50:14.590362 I | etcdserver: name = testname0
../bin/etcd-23083: 2017-10-25 13:50:14.590377 I | etcdserver: data dir = /tmp/testname0.etcd342488895
../bin/etcd-23083: 2017-10-25 13:50:14.590381 I | etcdserver: member dir = /tmp/testname0.etcd342488895/member
../bin/etcd-23083: 2017-10-25 13:50:14.590387 I | etcdserver: heartbeat = 100ms
../bin/etcd-23083: 2017-10-25 13:50:14.590392 I | etcdserver: election = 1000ms
../bin/etcd-23083: 2017-10-25 13:50:14.590397 I | etcdserver: snapshot count = 100000
../bin/etcd-23083: 2017-10-25 13:50:14.590430 I | etcdserver: advertise client URLs = https://localhost:20000
../bin/etcd-23083: 2017-10-25 13:50:14.590440 I | etcdserver: initial advertise peer URLs = http://localhost:20001
../bin/etcd-23083: 2017-10-25 13:50:14.590451 I | etcdserver: initial cluster = testname0=http://localhost:20001
../bin/etcd-23083: 2017-10-25 13:50:14.594078 I | etcdserver: starting member ca50e9357181d758 in cluster 34f27e83b3bc2ff
../bin/etcd-23083: 2017-10-25 13:50:14.594100 I | raft: ca50e9357181d758 became follower at term 0
../bin/etcd-23083: 2017-10-25 13:50:14.594116 I | raft: newRaft ca50e9357181d758 [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
../bin/etcd-23083: 2017-10-25 13:50:14.594126 I | raft: ca50e9357181d758 became follower at term 1
../bin/etcd-23083: 2017-10-25 13:50:14.599515 W | auth: simple token is not cryptographically signed
../bin/etcd-23083: 2017-10-25 13:50:14.601639 I | etcdserver: starting server... [version: 3.2.0+git, cluster version: to_be_decided]
../bin/etcd-23083: 2017-10-25 13:50:14.601781 I | embed: ClientTLS: cert = ../integration/fixtures/server.crt, key = ../integration/fixtures/server.key.insecure, ca = ../integration/fixtures/ca.crt, trusted-ca = , client-cert-auth = false, crl-file =
../bin/etcd-23083: 2017-10-25 13:50:14.602132 I | etcdserver/membership: added member ca50e9357181d758 [http://localhost:20001] to cluster 34f27e83b3bc2ff
../bin/etcd-23083: 2017-10-25 13:50:15.194415 I | raft: ca50e9357181d758 is starting a new election at term 1
../bin/etcd-23083: 2017-10-25 13:50:15.194453 I | raft: ca50e9357181d758 became candidate at term 2
../bin/etcd-23083: 2017-10-25 13:50:15.194522 I | raft: ca50e9357181d758 received MsgVoteResp from ca50e9357181d758 at term 2
../bin/etcd-23083: 2017-10-25 13:50:15.194549 I | raft: ca50e9357181d758 became leader at term 2
../bin/etcd-23083: 2017-10-25 13:50:15.194558 I | raft: raft.node: ca50e9357181d758 elected leader ca50e9357181d758 at term 2
../bin/etcd-23083: 2017-10-25 13:50:15.194750 I | etcdserver: setting up the initial cluster version to 3.2
../bin/etcd-23083: 2017-10-25 13:50:15.195988 N | etcdserver/membership: set the initial cluster version to 3.2
../bin/etcd-23083: 2017-10-25 13:50:15.196042 I | etcdserver/api: enabled capabilities for version 3.2
../bin/etcd-23083: 2017-10-25 13:50:15.196065 I | embed: ready to serve client requests
../bin/etcd-23083: 2017-10-25 13:50:15.196086 I | etcdserver: published {Name:testname0 ClientURLs:[https://localhost:20000]} to cluster 34f27e83b3bc2ff
../bin/etcd-23083: 2017-10-25 13:50:15.196183 E | etcdmain: forgot to set Type=notify in systemd service file?
../bin/etcd-23083: 2017-10-25 13:50:15.198057 I | embed: serving client requests on 127.0.0.1:20000
../bin/etcdctl-23096: OK
../bin/etcdctl-23095: PUT
../bin/etcdctl-23095: sample
../bin/etcdctl-23095: value
../bin/etcd-23083: panic: send on closed channel
../bin/etcd-23083:
../bin/etcd-23083: goroutine 122 [running]:
../bin/etcd-23083: github.com/coreos/etcd/cmd/vendor/google.golang.org/grpc/transport.(*serverHandlerTransport).do(0xc420175880, 0xc42068d1e0, 0x0, 0x0)
../bin/etcd-23083: 	/home/gyuho/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/cmd/vendor/google.golang.org/grpc/transport/handler_server.go:170 +0x115
../bin/etcd-23083: github.com/coreos/etcd/cmd/vendor/google.golang.org/grpc/transport.(*serverHandlerTransport).WriteStatus(0xc420175880, 0xc42017e600, 0xc4205b2040, 0xc420171110, 0xfd40a0)
../bin/etcd-23083: 	/home/gyuho/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/cmd/vendor/google.golang.org/grpc/transport/handler_server.go:187 +0xbd
../bin/etcd-23083: github.com/coreos/etcd/cmd/vendor/google.golang.org/grpc.(*serverStream).RecvMsg.func1(0xc4205c01b0, 0xc420605cc8, 0xefdc80, 0xc4201f0300)
../bin/etcd-23083: 	/home/gyuho/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/cmd/vendor/google.golang.org/grpc/stream.go:654 +0xb0
../bin/etcd-23083: github.com/coreos/etcd/cmd/vendor/google.golang.org/grpc.(*serverStream).RecvMsg(0xc4205c01b0, 0xefdc80, 0xc4201f0300, 0x1556260, 0xc420171110)
../bin/etcd-23083: 	/home/gyuho/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/cmd/vendor/google.golang.org/grpc/stream.go:668 +0x170
../bin/etcd-23083: github.com/coreos/etcd/cmd/vendor/github.com/coreos/etcd/etcdserver/api/v3rpc.(*serverStreamWithCtx).RecvMsg(0xc42022acf0, 0xefdc80, 0xc4201f0300, 0x10, 0xe5edc0)
../bin/etcd-23083: 	<autogenerated>:1 +0x50
../bin/etcd-23083: github.com/coreos/etcd/cmd/vendor/github.com/grpc-ecosystem/go-grpc-prometheus.(*monitoredServerStream).RecvMsg(0xc42068c0a0, 0xefdc80, 0xc4201f0300, 0xc0b66a, 0xc42060a001)
../bin/etcd-23083: 	/home/gyuho/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/cmd/vendor/github.com/grpc-ecosystem/go-grpc-prometheus/server.go:69 +0x4b
../bin/etcd-23083: github.com/coreos/etcd/cmd/vendor/github.com/coreos/etcd/etcdserver/etcdserverpb.(*watchWatchServer).Recv(0xc4201f01d0, 0x0, 0x0, 0x0)
../bin/etcd-23083: 	/home/gyuho/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/cmd/vendor/github.com/coreos/etcd/etcdserver/etcdserverpb/rpc.pb.go:3499 +0x62
../bin/etcd-23083: github.com/coreos/etcd/cmd/vendor/github.com/coreos/etcd/etcdserver/api/v3rpc.(*serverWatchStream).recvLoop(0xc4201ba6e0, 0x0, 0x0)
../bin/etcd-23083: 	/home/gyuho/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/cmd/vendor/github.com/coreos/etcd/etcdserver/api/v3rpc/watch.go:175 +0x4c
../bin/etcd-23083: github.com/coreos/etcd/cmd/vendor/github.com/coreos/etcd/etcdserver/api/v3rpc.(*watchServer).Watch.func2(0xc4201ba6e0, 0xc42021ecc0)
../bin/etcd-23083: 	/home/gyuho/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/cmd/vendor/github.com/coreos/etcd/etcdserver/api/v3rpc/watch.go:142 +0x2b
../bin/etcd-23083: created by github.com/coreos/etcd/cmd/vendor/github.com/coreos/etcd/etcdserver/api/v3rpc.(*watchServer).Watch
../bin/etcd-23083: 	/home/gyuho/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/cmd/vendor/github.com/coreos/etcd/etcdserver/api/v3rpc/watch.go:141 +0x2e5
../bin/etcdctl-23115: Error: dial tcp 127.0.0.1:20000: getsockopt: connection refused
../bin/etcdctl-23116: Error: dial tcp 127.0.0.1:20000: getsockopt: connection refused
--- FAIL: TestCtlV3WatchClientTLS (15.62s)
	ctl_v3_watch_test.go:82: watchTest #1: ctlV3Watch error (read /dev/ptmx: input/output error)
	ctl_v3_watch_test.go:75: watchTest #1-0: ctlV3Put error (read /dev/ptmx: input/output error (expected "OK", got ["Error: dial tcp 127.0.0.1:20000: getsockopt: connection refused\r\n"]))
	testutil.go:55: goroutine 14 [running]:
		github.com/coreos/etcd/pkg/testutil.FatalStack(0xc4202141e0, 0xc420246400, 0x18)
			/home/gyuho/go/src/github.com/coreos/etcd/pkg/testutil/testutil.go:54 +0x74
		github.com/coreos/etcd/e2e.testCtl(0xc4202141e0, 0xcc1068, 0xc420047f90, 0x1, 0x1)
			/home/gyuho/go/src/github.com/coreos/etcd/e2e/ctl_v3_test.go:164 +0x483
		github.com/coreos/etcd/e2e.TestCtlV3WatchClientTLS(0xc4202141e0)
			/home/gyuho/go/src/github.com/coreos/etcd/e2e/ctl_v3_watch_test.go:24 +0xa9
		testing.tRunner(0xc4202141e0, 0xcc0c60)
			/usr/local/go/src/testing/testing.go:746 +0xd0
		created by testing.(*T).Run
			/usr/local/go/src/testing/testing.go:789 +0x2de

		goroutine 1 [chan receive]:
		testing.(*T).Run(0xc4202140f0, 0xc9d96b, 0x17, 0xcc0c60, 0x48f501)
			/usr/local/go/src/testing/testing.go:790 +0x2fc
		testing.runTests.func1(0xc4202140f0)
			/usr/local/go/src/testing/testing.go:1004 +0x64
		testing.tRunner(0xc4202140f0, 0xc420211d98)
			/usr/local/go/src/testing/testing.go:746 +0xd0
		testing.runTests(0xc420159a20, 0x10ad740, 0x97, 0x97, 0xc420022767)
			/usr/local/go/src/testing/testing.go:1002 +0x2d8
		testing.(*M).Run(0xc420211f18, 0xc9d674)
			/usr/local/go/src/testing/testing.go:921 +0x111
		github.com/coreos/etcd/e2e.TestMain(0xc420211f18)
			/home/gyuho/go/src/github.com/coreos/etcd/e2e/main_test.go:52 +0x4d5
		main.main()
			github.com/coreos/etcd/e2e/_test/_testmain.go:342 +0xdb

		goroutine 42 [chan receive]:
		github.com/coreos/etcd/pkg/logutil.(*MergeLogger).outputLoop(0xc420181c00)
			/home/gyuho/go/src/github.com/coreos/etcd/pkg/logutil/merge_logger.go:174 +0x47a
		created by github.com/coreos/etcd/pkg/logutil.NewMergeLogger
			/home/gyuho/go/src/github.com/coreos/etcd/pkg/logutil/merge_logger.go:92 +0xb8

		goroutine 51 [chan receive]:
		github.com/coreos/etcd/e2e.watchTest(0xc4202141e0, 0xc420014be0, 0xb, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, 0x4e20, ...)
			/home/gyuho/go/src/github.com/coreos/etcd/e2e/ctl_v3_watch_test.go:85 +0x5fe
		github.com/coreos/etcd/e2e.testCtl.func2(0xc420238000, 0xcc1068, 0xc4202142d0)
			/home/gyuho/go/src/github.com/coreos/etcd/e2e/ctl_v3_test.go:155 +0x83
		created by github.com/coreos/etcd/e2e.testCtl
			/home/gyuho/go/src/github.com/coreos/etcd/e2e/ctl_v3_test.go:153 +0x2c5

	testutil.go:56: test timed out after 15s
	ctl_v3_test.go:148: error closing etcd processes (exit status 2)
FAIL
exit status 1
FAIL	github.com/coreos/etcd/e2e	15.628s
```

**Comment by xiang90** [comment]
closing.
